tofhir-server/src/main/scala/io/tofhir/server/endpoint/JobEndpoint.scala

```diff
@@ -42,7 +42,25 @@ class JobEndpoint(jobRepository: IJobRepository, mappingRepository: IMappingRepo
           }
         } ~ pathPrefix(SEGMENT_TEST) { // test a mapping with mapping job configurations, jobs/<jobId>/test
           pathEndOrSingleSlash {
-            testMappingWithJob(projectId, jobId)
+            withRequestTimeoutResponse { _ =>
+              addCORSHeaders(
+                HttpResponse(
+                  status = StatusCodes.RequestTimeout,
+                  entity = HttpEntity.Strict(
+                    ContentTypes.`application/json`,
+                    ByteString(
+                      """Status Code: 408
```

Comment:
Instead of writing plain string here, can you add a new case class to ToFhirError file for timeout errors. And use that class to construct the error.